Makefile

```diff
@@ -1,123 +1,93 @@
-# Metadata about this makefile and position
-MKFILE_PATH := $(lastword $(MAKEFILE_LIST))
-CURRENT_DIR := $(patsubst %/,%,$(dir $(realpath $(MKFILE_PATH))))
+SHELL := /bin/bash -o pipefail
 
-# Ensure GOPATH
-GOPATH ?= $(HOME)/go
+# List of tests to run
+FILES ?= ./...
 
 # List all our actual files, excluding vendor
-GOFILES ?= $(shell go list $(FILES) | grep -v /vendor/)
+GOPKGS ?= $(shell go list $(FILES) | grep -v /vendor/)
+GOFILES ?= $(shell find . -name '*.go' | grep -v /vendor/)
 
 # Tags specific for building
 GOTAGS ?=
 
 # Number of procs to use
 GOMAXPROCS ?= 4
 
-PROJECT := $(CURRENT_DIR:$(GOPATH)/src/%=%)
-OWNER := $(notdir $(patsubst %/,%,$(dir $(PROJECT))))
-NAME := $(notdir $(PROJECT))
-EXTERNAL_TOOLS = \
-	github.com/golang/dep/cmd/dep
-
-# Current system information
-GOOS ?= $(shell go env GOOS)
-GOARCH ?= $(shell go env GOARCH)
-GOCACHE ?= $(shell go env GOCACHE)
-
-# List of tests to run
-FILES ?= ./...
+NAME := $(notdir $(shell pwd))
 
 # Test Service ID
 FASTLY_TEST_SERVICE_ID ?=
 
-# bootstrap installs the necessary go tools for development or build.
-bootstrap:
-	@echo "==> Bootstrapping ${PROJECT}"
-	@for t in ${EXTERNAL_TOOLS}; do \
-		echo "--> Installing $$t" ; \
-		go get -u "$$t"; \
-	done
-.PHONY: bootstrap
-
-clean:
-	@echo "==> Cleaning ${NAME}"
-	@rm -rf pkg
-.PHONY: clean
-
-# build builds the binary into pkg/
-build:
-	@echo "==> Building ${NAME} for ${GOOS}/${GOARCH}"
-	@env \
-		-i \
-		PATH="${PATH}" \
-		CGO_ENABLED="0" \
-		GOOS="${GOOS}" \
-		GOARCH="${GOARCH}" \
-		GOCACHE="${GOCACHE}" \
-		GOPATH="${GOPATH}" \
-		go build -a -o "pkg/${GOOS}_${GOARCH}/${NAME}" ${GOFILES}
-.PHONY: build
-
-# deps updates all dependencies for this project.
-deps:
-	@echo "==> Updating deps for ${PROJECT}"
-	@dep ensure -update
-	@dep prune
-.PHONY: deps
-
-# dev builds and installs the
-dev:
-	@echo "==> Installing ${NAME} for ${GOOS}/${GOARCH}"
-	@env \
-		-i \
-		PATH="${PATH}" \
-		CGO_ENABLED="0" \
-		GOOS="${GOOS}" \
-		GOARCH="${GOARCH}" \
-		GOCACHE="${GOCACHE}" \
-		GOPATH="${GOPATH}" \
-		go install ${GOFILES}
-.PHONY: dev
-
-# linux builds the linux binary
-linux:
-	@env \
-		GOOS="linux" \
-		GOARCH="amd64" \
-		$(MAKE) -f "${MKFILE_PATH}" build
-.PHONY: linux
+all: tidy fmt fiximports test vet staticcheck ## Runs all of the required cleaning and verification targets.
```

Comment:
Maybe we should add a target to install the tooling required to run these tasks, some people may not have `goimport` and `staticcheck` installed.

Reply:
So `tidy` would install these tools, right, because they are part of the Go module and `go` commands would pull down dependencies.

I may be wrong. Maybe a `download` target that does a `go mod download`.

Reply:
I was wrong.

```bash
$ rm -rf $(which staticcheck)
$ make
```

Fails with:

```bash
==> Running staticcheck
/bin/bash: staticcheck: command not found
make: *** [Makefile:80: staticcheck] Error 127
```

Adding the `download` target.

Reply:
Ahh I was still wrong. This would just allow versioning of the tools, but doesn't actually install the binaries. So, yes, you were correct, we would also need an "install" target. I'm going to call it `dev-dependencies`.

Reply:
Confirmed that it worked:

```bash
$ rm -rf $(which goimports)
$ rm -rf $(which staticcheck)
$ make
==> Downloading Go module
==> Downloading development dependencies
==> Tidying module
==> Running gofmt
==> Fixing imports
==> Testing go-fastly
ok      github.com/fastly/go-fastly/fastly      (cached)
==> Running go vet
==> Running staticcheck
make  7.83s user 0.82s system 208% cpu 4.154 total
```